BootloaderCorePkg/Stage2/Stage2.c

```diff
@@ -119,8 +119,9 @@ PreparePayload (
       }
     }
 
-    if (IsNormalPld || (Stage2Hob->PayloadId != 0)) {
+    if (IsNormalPld || (Stage2Hob->PayloadId != 0) || (BootMode == BOOT_ON_FLASH_UPDATE)) {
```

Comment:
I think it is better to set IsNomralPld to TRUE when firmware update is detected.